src/flexbuffer_json.h

```diff
@@ -40,27 +40,27 @@ struct JsonPath {
             if( idx < std::numeric_limits<index_type>::max() ) {
                 path_.push_back( static_cast<index_type>( idx ) );
             } else {
-                throw std::runtime_error( "Json index out of range of uint16_t" );
+                throw std::runtime_error( "Json index out of range of uint32_t" );
             }
             return *this;
         }
 
-        uint16_t size() const {
+        uint8_t size() const {
```

Comment:
This doesn't seem quite right, or am I missing something?

Reply:
The size this is referring to is the "depth" of the Json object from the root. Eg. the deepest thing in `{"x": [1, {"y": 2, "z": [3]}]}` is at depth 4 with a JsonPath that would look like  `[0, 1, 1, 0]`. I tried to make it `decltype(path_.size())` but trailing return type syntax has weird restrictions on what members are in scope or not and I didn't feel like typing out the whole type. uint8_t is the default size and I'm hoping compiler flags will flag if that ever changes because the underlying size type in `path_` changes.